sound/soc/sof/ipc4-topology.c

```diff
@@ -1659,11 +1638,7 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 	}
 
 	/* modify the input params for the next widget */
-	fmt = hw_param_mask(pipeline_params, SNDRV_PCM_HW_PARAM_FORMAT);
-	out_sample_valid_bits =
-		SOF_IPC4_AUDIO_FORMAT_CFG_V_BIT_DEPTH(copier_data->out_format.fmt_cfg);
-	snd_mask_none(fmt);
-	ret = ipc4_set_fmt_mask(fmt, out_sample_valid_bits);
+	ret = sof_ipc4_update_hw_params(sdev, pipeline_params, &copier_data->out_format);
```

Comment:
not able to follow this at all. 

There should be nothing that "modifies the pipeline_params() at the output of the copier", that can only happen at the output of the SRC.

Reply:
@plbossart its not that the copier modifes the rate. The problem is pipeline_params starts off being the same as fe_params and the copier must set to the righ value based on what its input/output rate is. I've modified the commit message now. Hope it makes sense